compiler/src/dmd/iasmgcc.d

```diff
@@ -252,7 +253,7 @@ int parseExtAsmOperands(Parser)(Parser p, GccAsmStatement s)
             }
             else
             {
-                p.eSink.error(p.token.loc, "expected identifier after `[`");
+                p.eSink.error(p.token.loc, "identifier expected after `[`");
```

Comment:
parsePrimary issues `expression expected`, so swap for consistency.  Order doesn't matter in English.